tests/unit/test-extension-runners/test.extension-runners.firefox-desktop.js

```diff
@@ -0,0 +1,407 @@
+/* @flow */
+
+import {describe, it} from 'mocha';
+import {assert} from 'chai';
+import sinon from 'sinon';
+
+import {
+  FirefoxDesktopExtensionRunner,
+} from '../../../src/extension-runners/firefox-desktop';
+import {
+  getFakeFirefox,
+  getFakeRemoteFirefox,
+  makeSureItFails,
+  StubChildProcess,
+} from '../helpers';
+import {
+  onlyInstancesOf,
+  WebExtError,
+  RemoteTempInstallNotSupported,
+} from '../../../src/errors';
+
+// Fake result for client.installTemporaryAddon().then(installResult => ...)
+const tempInstallResult = {
+  addon: {id: 'some-addon@test-suite'},
+};
+// Fake missing addon id result for client.installTemporaryAddon
+const tempInstallResultMissingAddonId = {
+  addon: {id: null},
+};
+
+type PrepareParams = {
+  params?: Object,
+  deps?: Object,
+  fakeFirefoxApp?: Object,
+  fakeRemoteFirefox?: Object,
+  debuggerPort?: number,
+}
+
+function prepareExtensionRunnerParams({
+  params, fakeFirefoxApp, fakeRemoteFirefox, debuggerPort,
+}: PrepareParams = {}) {
+  const remoteFirefox = getFakeRemoteFirefox({
+    installTemporaryAddon: sinon.spy(
+      () => Promise.resolve(tempInstallResult)
+    ),
+    ...fakeRemoteFirefox,
+  });
+  const firefoxProcess = new StubChildProcess();
+
+  return {
+    remoteFirefox,
+    firefoxProcess,
+    params: {
+      extensions: [{
+        sourceDir: '/fake/sourceDir',
+        manifestData: {
```

Comment:
I think it would be helpful to extend this from [`basicManifest`](https://github.com/mozilla/web-ext/blob/master/tests/unit/test-util/test.manifest.js#L15), at least just for documentation purposes.